index.ts

```diff
@@ -415,7 +415,32 @@ app.get("/favorites/:characterId", (req, res) => {
 
 app.get("/blacklist", (req, res) => {
     // e.g. http://localhost:3000/blacklist
-    res.render("blacklist");
+    if (user == null) {
+        return res.status(404).send("User not found");
+    }
+    const blacklist: Blacklist[] = user.blacklist;
+    res.render("blacklist", {blacklist: blacklist});
+})
+
+app.post("/blacklist/:quoteId/delete", (req, res) => {
+    if (user == null) {
+        return res.status(404).send("User not found");
+    }
+
+    const quoteId = req.params.quoteId;
+    deleteBlacklist(user, quoteId);
+    loadUser(user.username);
+})
```

Comment:
You have to redirect back to the page at the end of the route, otherwise it cannot escape:

` res.redirect("/blacklist");`